arangod/Graph/ShortestPathOptions.cpp

```diff
@@ -162,8 +160,13 @@ void ShortestPathOptions::addReverseLookupInfo(aql::ExecutionPlan* plan,
 
 double ShortestPathOptions::weightEdge(VPackSlice edge) const {
   TRI_ASSERT(useWeight());
-  return arangodb::basics::VelocyPackHelper::getNumericValue<double>(
-      edge, weightAttribute.c_str(), defaultWeight);
+  const auto weight = arangodb::basics::VelocyPackHelper::getNumericValue<double>(
+      edge, _weightAttribute.c_str(), _defaultWeight);
+  if (weight < 0.) {
```

Comment:
```suggestion
  if (weight <= 0.) {
```

Reply:
TY. will apply in separate commit as I am lazy and want to cherry-pick in a single commit :)